apis/otpapi/otpapi.yaml

```diff
@@ -0,0 +1,210 @@
+host: staging.open-sunbird.org
+basePath:  /api/otp/v1
+securityDefinitions:
+  bearer:
+      type: apiKey  # arbitrary name for the security scheme
+      in: header # The secrity scheme parameter is in "header"
+      name: Authorization # name of the header, query parameter or cookie
+schemes:
+  - https
+consumes:
+  - application/json
+produces:
+  - application/json
+paths:
+  /generate:
+    post:
+      description: >-
+       This API is associated with generating OTP on Sunbird platform.
+          - The **_/otp/v1/generate_** endpoint executes the "Generate OTP" request based on parameters provided as metadata in the request body
+          - Required fields cannot be left null or empty
+      summary: Generate OTP
+      tags: 
+      - OTP APIs
+      operationId: GenerateOTP
+      produces:
+      - application/json
+      parameters:
+      - name: Body
+        in: body
+        required: true
+        description: >-
+            The body refers to the format of the request.
+              - The body contains metadata about the request for generating OTP.
+        schema:  
+          $ref: '#/definitions/GenerateOTPRequest'
+      - name: Content-Type
+        in: header
+        required: true
+        type: string
+        description: >-
+           Type of the request data. Set as application/json.
+      - name: ts
+        in: header
+        required: true
+        type: string
+        description: 'Timestamp at which given API request is sent.'
+      - name: X-msgid
+        in: header
+        required: true
+        type: string
+        description: 'This ID uniquely identifies a request if the same API is executed multiple times.'
+      - name: Authorization
+        in: header
+        required: true
+        type: string
+        description: 'Specify authorization key (format: Bearer api-key) received from administrator when performing given API request.'
+      responses:
+        '200':
+          description: 'OK. Successful operation. The requested operation was successfully executed.'
+          schema:  
+            $ref: '#/definitions/CommonSuccessResponse'
+        '400':
+          description: 'CLIENT_ERROR. The requested operation failed due to bad request from client. Possible reasons for failure:
+              -  Mandatory parameter is missing.
+              -  Email is invalid.
+              -  Please provide a valid phone number.
+              -  Invalid type. Valid values are: email, phone.'
+          schema:  
+            $ref: '#/definitions/CommonErrorResponse'
+        '429':
+          description: 'TOO_MANY_REQUESTS. The requested operation failed. Possible reasons for failure:
+              - Your per hour/day rate limit has exceeded. You can retry after some time.'
+          schema:  
+            $ref: '#/definitions/CommonErrorResponse'    
+        '500':
+          description: 'INTERNAL SERVER ERROR. The requested operation failed due to a server error.'
```

Comment:
INTERNAL SERVER ERROR, we are tracking the error and cater to it on priority. Try to refresh the page and if the problem persists contact us at info@sunbird.org